src/Security/SecProtocolMetadata.cs

```diff
@@ -51,9 +51,173 @@ public class SecProtocolMetadata : NativeObject {
 
 		public bool EarlyDataAccepted => sec_protocol_metadata_get_early_data_accepted (GetCheckedHandle ()) != 0;
 
-		//
-		// MISSING: all the block APIs
-		//
+ 		[DllImport (Constants.SecurityLibrary)]
+		[return: MarshalAs (UnmanagedType.I1)]
+		extern static bool sec_protocol_metadata_challenge_parameters_are_equal (IntPtr metadataA, IntPtr metadataB);
 
+		public static bool ParametersAreEqual (SecProtocolMetadata metadataA, SecProtocolMetadata metadataB)
+		{
+			return sec_protocol_metadata_challenge_parameters_are_equal (metadataA.GetCheckedHandle (), metadataB.GetCheckedHandle ());
```

Comment:
there's no null check on parameters -> NRE
I think a better name would be `ChallengeParametersAreEqual`

even if I'd prefer `CompareChallengeParameters` but that suggest returning an `int` :|

Reply:
The headers do not mention anything about null, so I'll add the null check.

Reply:
@spouliot what about just `Equals` (like `string.Equals`)? The input parameter types varies, so there can be multiple overloads with that name.